studio/components/interfaces/Auth/SmtpForm.tsx

```diff
@@ -0,0 +1,338 @@
+import { Form, Input, Toggle } from '@supabase/ui'
+import { useStore } from 'hooks'
+import { observer } from 'mobx-react-lite'
+import { useEffect, useState } from 'react'
+import { number, object, string } from 'yup'
+import {
+  FormActions,
+  FormHeader,
+  FormPanel,
+  FormSection,
+  FormSectionContent,
+  FormSectionLabel,
+} from '../../ui/Forms'
+
+const SmtpForm = observer(() => {
+  const { authConfig, ui } = useStore()
+  const [enableSmtp, setEnableSmtp] = useState(false)
+
+  function generateInitialValues(config: any) {
+    return {
+      ENABLE_SMTP: enableSmtp,
+      SMTP_ADMIN_EMAIL: config.SMTP_ADMIN_EMAIL ?? '',
+      SMTP_SENDER_NAME: config.SMTP_SENDER_NAME ?? '',
+      SMTP_USER: config.SMTP_USER ?? '',
+      SMTP_HOST: config.SMTP_HOST ?? '',
+      SMTP_PASS: config.SMTP_PASS ?? '',
+      SMTP_PORT: config.SMTP_PORT ?? 465,
+      SMTP_MAX_FREQUENCY: config.SMTP_MAX_FREQUENCY ?? undefined,
+    }
+  }
+
+  function checkSmtpState(config: any, resetForm: () => void) {
+    if (
+      config.SMTP_ADMIN_EMAIL &&
+      config.SMTP_SENDER_NAME &&
+      config.SMTP_USER &&
+      config.SMTP_HOST &&
+      config.SMTP_PASS &&
+      config.SMTP_PORT &&
+      config.SMTP_MAX_FREQUENCY > 0
+    ) {
+      setEnableSmtp(true)
+      return resetForm
+    } else {
+      setEnableSmtp(false)
+      return resetForm
+    }
+  }
+
+  /**
+   * domain and IP address regex
+   *
+   * example of matches:
+   *
+   * "vercel.com"
+   * "www.vercel.com"
+   * "uptime-monitor-fe.vercel.app"
+   * "https://uptime-monitor-fe.vercel.app/"
+   * "127.0.0.0"
+   *
+   */
+  const domainRegex =
```

Comment:
chuck this into an Auth.constants.ts? since used in RedirectDomains too

Reply:
its actually slightly different i think

Reply:
one of them is also checking for IP address